File: data/tools.py
from ..core.builder import *
from ..core.images import loader
from ..core.structure import Shape
from ..core.util import extract

from .lib.items import *


def init():
    gervais = loader('icons/gervais_roguelike')

    mk_item('pick', 'Pickaxe', gervais('AngbandTk_pick.png')) \
            .recipe('anvil', {'wood': 10, 'stone': 10}, count=5)
    mk_item('axe', 'Axe', gervais('AngbandTk_axe.png')) \
            .recipe('anvil', {'wood': 10, 'stone': 10})
    mk_item('mallet', 'Mallet', gervais('AngbandTk_mallet.png')) \
            .recipe('anvil', {'wood': 20})
    mk_item('shovel', 'Shovel', gervais('AngbandTk_shovel.png')) \
            .recipe('anvil', {'wood': 10, 'stone': 10})

File: scripts/outpost/tools.lua
local action = require('core.action')
local mallet = require('outpost.lib.mallet')
local tools = require('outpost.lib.tools')

action.use_item.mallet = mallet.use_mallet
action.use_item.axe = tools.mk_use_tool('axe')
action.use_item.pick = tools.mk_use_tool('pick')
